Sources/Sentry/SentryClient.m

```diff
@@ -422,6 +433,35 @@ - (void)setUserIdIfNoUserSet:(SentryEvent *)event
     }
 }
 
+/**
+ * Calls onCrashedLastRun if the execution terminated with a crash.
+ */
+- (void)callOnCrashedLastRun:(SentryEvent *)event
+{
+    if (nil != self.options.onCrashedLastRun && self.crashAdapter.crashedLastLaunch
+        && !SentrySDK.crashedLastRunCalled && nil != event.exceptions) {
+
+        NSPredicate *unhandledExpeptions = [NSPredicate predicateWithBlock:^BOOL(
+            id _Nullable object, NSDictionary<NSString *, id> *_Nullable bindings) {
+            SentryException *exception = object;
+            return nil != exception.mechanism && nil != exception.mechanism.handled
+                && ![exception.mechanism.handled boolValue];
+        }];
+
+        // We only want to filter the array if the above conditions are true to avoid unecessary
+        // work
+        BOOL eventContainsUnhandledExceptions =
+            [event.exceptions filteredArrayUsingPredicate:unhandledExpeptions].count > 0;
+
+        if (eventContainsUnhandledExceptions) {
+            // We only want to call the callback once. It can occur that multiple crash events are
+            // about to be sent.
+            SentrySDK.crashedLastRunCalled = YES;
+            self.options.onCrashedLastRun(event);
```

Comment:
```suggestion
            self.options.onCrashedLastRun(event);
            SentrySDK.crashedLastRunCalled = YES;
```

Reply:
L: no strong feelings here, but I'm used to seeing booleans being toggled after the action, not before.